autoPyTorch/datasets/base_dataset.py

```diff
@@ -312,22 +324,30 @@ def create_holdout_val_split(
             self.random_state, val_share, self._get_indices(), **kwargs)
         return train, val
 
-    def get_dataset_for_training(self, split_id: int) -> Tuple[Dataset, Dataset]:
+    def get_dataset_for_training(self, split_id: int, train: bool) -> Dataset:
```

Comment:
Since it is not only `_for_training`, we should rename it.
```suggestion
    def get_dataset(self, split_id: int, train: bool) -> Dataset:
```